concent_api/gatekeeper/views.py

```diff
@@ -0,0 +1,72 @@
+from base64                         import b64decode
+
+from django.conf                    import settings
+from django.http                    import JsonResponse, HttpResponse
+from django.views.decorators.http   import require_POST, require_GET
+from django.views.decorators.csrf   import csrf_exempt
+
+from golem_messages.shortcuts       import load
+from golem_messages.message         import Message, init_messages
+
+
+@csrf_exempt
+@require_POST
+def upload(request):
+    if request.content_type != 'multipart/form-data':
+        return json_response('Wrong upload data content type.')
+    init_messages()
+    check_header = check_headers(request)
+    if check_header is not None:
+        return check_header
+
+    return HttpResponse("", status=200)
+
+
+@csrf_exempt
+@require_GET
+def download(request):
+    if request.content_type == 'multipart/form-data':
+        return json_response('Wrong download data content type.')
+    init_messages()
+    check_header = check_headers(request)
+    if check_header is not None:
+        return check_header
+
+    return HttpResponse("", status=200)
+
+
+def check_headers(request):
+    # Decode and check if request header contains golem message:
+    if 'HTTP_AUTHORIZATION' not in request.META:
+        return json_response('Missing Authorization header.')
+
+    authorization = request.META['HTTP_AUTHORIZATION']
+
+    if len(authorization.split(" ")) != 2:
+        return json_response('Wrong construction of Authorization header.')
+    (golem_header, auth_header_content) = authorization.split()
+
+    if golem_header != 'Golem':
+        return json_response('Authorization header should start with "Golem".')
+
+    try:
+        decoded_auth_header_content = (b64decode(auth_header_content))
+    except ValueError:
+        return json_response('Unable to decode Authorization header.')
+
+    try:
+        loaded_golem_msg = load(decoded_auth_header_content, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+    except AttributeError:
+        return json_response('Authorization header is not a golem message.')
+
+    if loaded_golem_msg is None:
+        return json_response('Undefined golem message.')
+    assert isinstance(loaded_golem_msg, Message)
+
+
+
+def json_response(message):
```

Comment:
Please move this function to utils or something similar.

Reply:
I've made gatekeeper_view.py in utils and put it there

Reply:
I don't think it's generic enough to warrant moving it to `utils`. It should have much more specific name. `json_response()` does not give any hint that it's hard-coded to return HTTP 401 and add `WWW-Authenticate` header. By looking at the call you can't even tell that it will return an error. It should be something like `gatekeeper_access_denied_response()`. And in my opinion this is something very specific to gatekeeper, not a generic utility that could be used by other apps.

Reply:
I agree but I still don't think it should be in views.py. How about putting it into gatekeeper/utils.py or gatekeeper/helpers.py ?

Reply:
Personally, I'd keep it in views (maybe as a nested function in the view?) because it's been made into a function mostly to save repetition. It's not really reusable outside of that single view.

But well, as long as it's inside the gatekeeper app it does not matter that much so if you stronly prefer to put it in some kind of `utils` file, I'm fine with that.

Reply:
From my experience when there is permission to put objects that are not strictly elements that should be in that files (like views in views, models in models etc.) sooner or later there is more and more of those in that files and it is getting more spaghetti. I suggest putting it into separate file in that app, or, if it is reusable, use ClassBasedViews with Base class with that function as a method, and inherit it in other views.

Reply:
Done, switched to `gatekeeper_access_denied_response()` in  gatekeeper/utils.py